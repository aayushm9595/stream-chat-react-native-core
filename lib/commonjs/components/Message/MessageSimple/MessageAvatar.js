var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageAvatar = void 0;

var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));

var _react = _interopRequireDefault(require("react"));

var _Avatar = require("../../Avatar/Avatar");

var _styledComponents = require("../../../styles/styledComponents");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/Message/MessageSimple/MessageAvatar.tsx";

function _templateObject2() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  height: 28px;\n  width: 32px;\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  margin-left: ", "px;\n  margin-right: ", "px;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Container = _styledComponents.styled.View(_templateObject(), function (_ref) {
  var alignment = _ref.alignment;
  return alignment === 'right' ? 8 : 0;
}, function (_ref2) {
  var alignment = _ref2.alignment;
  return alignment === 'left' ? 8 : 0;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.message.avatarWrapper.container.css;
});

var Spacer = _styledComponents.styled.View(_templateObject2(), function (_ref4) {
  var theme = _ref4.theme;
  return theme.message.avatarWrapper.spacer.css;
});

var MessageAvatar = function MessageAvatar(props) {
  var _message$user, _message$user2, _message$user3;

  var alignment = props.alignment,
      groupStyles = props.groupStyles,
      message = props.message,
      showAvatar = props.showAvatar;
  var visible = typeof showAvatar === 'boolean' ? showAvatar : groupStyles[0] === 'single' || groupStyles[0] === 'bottom';
  return _react["default"].createElement(Container, {
    alignment: alignment,
    testID: "message-avatar",
    __source: {
      fileName: _jsxFileName,
      lineNumber: 64,
      columnNumber: 5
    }
  }, visible ? _react["default"].createElement(_Avatar.Avatar, {
    image: (_message$user = message.user) == null ? void 0 : _message$user.image,
    name: ((_message$user2 = message.user) == null ? void 0 : _message$user2.name) || ((_message$user3 = message.user) == null ? void 0 : _message$user3.id),
    __source: {
      fileName: _jsxFileName,
      lineNumber: 66,
      columnNumber: 9
    }
  }) : _react["default"].createElement(Spacer, {
    testID: "spacer",
    __source: {
      fileName: _jsxFileName,
      lineNumber: 71,
      columnNumber: 9
    }
  }));
};

exports.MessageAvatar = MessageAvatar;
//# sourceMappingURL=MessageAvatar.js.map