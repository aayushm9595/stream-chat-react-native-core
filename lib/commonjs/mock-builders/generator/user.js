var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateStaticUser = exports.generateUser = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _uuid = require("uuid");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var generateUser = function generateUser() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    banned: false,
    created_at: '2020-04-27T13:39:49.331742Z',
    id: (0, _uuid.v4)(),
    image: (0, _uuid.v4)(),
    name: (0, _uuid.v4)(),
    online: false,
    role: 'user',
    updated_at: '2020-04-27T13:39:49.332087Z'
  }, options);
};

exports.generateUser = generateUser;

var generateStaticUser = function generateStaticUser(userNumber) {
  switch (userNumber) {
    case 0:
      return generateUser({
        id: 'DannyBoi1',
        image: 'https://images-na.ssl-images-amazon.com/images/I/51t29lLkg8L._AC_SL1000_.jpg',
        name: 'Dan'
      });

    case 1:
      return generateUser({
        id: 'GrantyBoi2',
        image: 'https://i.imgur.com/SLx06PP.png',
        name: 'Grant'
      });

    case 2:
      return generateUser({
        id: 'ViriBoi2',
        image: 'https://i.imgur.com/iNaC3K7.jpg',
        name: 'Vir'
      });

    default:
      return generateUser({
        id: 'NeiliBoi3',
        image: 'https://i.imgur.com/T68W8nR_d.webp?maxwidth=728&fidelity=grand',
        name: 'Neil'
      });
  }
};

exports.generateStaticUser = generateStaticUser;
//# sourceMappingURL=user.js.map