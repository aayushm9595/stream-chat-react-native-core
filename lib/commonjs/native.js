Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registerNativeHandlers = exports.pickDocument = exports.pickImage = exports.NetInfo = void 0;

var fail = function fail() {
  throw Error('Native handler was not registered, you should import stream-chat-expo or stream-chat-react-native');
};

var NetInfo = {
  addEventListener: fail,
  fetch: fail
};
exports.NetInfo = NetInfo;
var pickImage = fail;
exports.pickImage = pickImage;
var pickDocument = fail;
exports.pickDocument = pickDocument;

var registerNativeHandlers = function registerNativeHandlers(handlers) {
  if (handlers.NetInfo) {
    exports.NetInfo = NetInfo = handlers.NetInfo;
  }

  if (handlers.pickImage) {
    exports.pickImage = pickImage = handlers.pickImage;
  }

  if (handlers.pickDocument) {
    exports.pickDocument = pickDocument = handlers.pickDocument;
  }
};

exports.registerNativeHandlers = registerNativeHandlers;
//# sourceMappingURL=native.js.map