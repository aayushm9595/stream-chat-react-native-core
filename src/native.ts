import type { NetInfoSubscription } from '@react-native-community/netinfo';

const fail = () => {
  throw Error(
    'Native handler was not registered, you should import stream-chat-expo or stream-chat-react-native',
  );
};

type NetInfo = {
  addEventListener: (
    listener: (isConnected: boolean) => void,
  ) => NetInfoSubscription | never;
  fetch: (requestedInterface?: string | undefined) => Promise<boolean> | never;
};

export let NetInfo: NetInfo = {
  addEventListener: fail,
  fetch: fail,
};

type PickImage = ({
  compressImageQuality,
  maxNumberOfFiles,
}: {
  compressImageQuality?: number;
  maxNumberOfFiles?: number;
}) => Promise<{ cancelled: boolean; images?: { uri: string }[] }> | never;
export let pickImage: PickImage = fail;

type PickDocument = ({
  maxNumberOfFiles,
}: {
  maxNumberOfFiles?: number;
}) =>
  | Promise<{
      cancelled: boolean;
      docs?: {
        name: string;
        size?: number | string;
        type?: string;
        uri?: string;
      }[];
    }>
  | never;
export let pickDocument: PickDocument = fail;

type Handlers = {
  NetInfo?: NetInfo;
  pickDocument?: PickDocument;
  pickImage?: PickImage;
};
export const registerNativeHandlers = (handlers: Handlers) => {
  if (handlers.NetInfo) {
    NetInfo = handlers.NetInfo;
  }

  if (handlers.pickImage) {
    pickImage = handlers.pickImage;
  }

  if (handlers.pickDocument) {
    pickDocument = handlers.pickDocument;
  }
};
