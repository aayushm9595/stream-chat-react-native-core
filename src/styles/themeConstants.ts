import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
import type { CSSObject } from 'styled-components';

export const BASE_FONT_SIZE = 16;

export const Colors = {
  danger: '#EDD8DD',
  light: '#EBEBEB',
  primary: '#006cff',
  secondary: '#111',
  textDark: 'rgba(0,0,0,1)',
  textGrey: 'rgba(0,0,0,0.5)',
  textLight: 'white',
  transparent: 'transparent',
};

export type CSS = {
  css?: CSSObject;
};

export type MarkdownStyle = Partial<{
  blockQuoteBar: StyleProp<ViewStyle>;
  blockQuoteSection: StyleProp<ViewStyle>;
  blockQuoteSectionBar: StyleProp<ViewStyle>;
  blockQuoteText: StyleProp<ViewStyle>;
  codeBlock: StyleProp<TextStyle>;
  del: StyleProp<TextStyle>;
  em: StyleProp<TextStyle>;
  heading: StyleProp<TextStyle>;
  heading1: StyleProp<TextStyle>;
  heading2: StyleProp<TextStyle>;
  heading3: StyleProp<TextStyle>;
  heading4: StyleProp<TextStyle>;
  heading5: StyleProp<TextStyle>;
  heading6: StyleProp<TextStyle>;
  hr: StyleProp<ViewStyle>;
  image: StyleProp<ImageStyle>;
  inlineCode: StyleProp<TextStyle>;
  link: StyleProp<TextStyle>;
  listItem: StyleProp<ViewStyle>;
  listItemBulletType: string;
  listItemButton: StyleProp<TextStyle>;
  listItemNumber: StyleProp<TextStyle>;
  listItemText: StyleProp<TextStyle>;
  mailTo: StyleProp<TextStyle>;
  paragraph: StyleProp<ViewStyle>;
  strong: StyleProp<TextStyle>;
  table: StyleProp<ViewStyle>;
  tableHeader: StyleProp<ViewStyle>;
  tableHeaderCell: StyleProp<ViewStyle>;
  tableRow: StyleProp<ViewStyle>;
  tableRowCell: StyleProp<ViewStyle>;
  tableRowLast: StyleProp<ViewStyle>;
  text: StyleProp<TextStyle>;
  u: StyleProp<TextStyle>;
  video: StyleProp<ImageStyle>;
  view: StyleProp<ViewStyle>;
}>;

export type Theme = {
  avatar: {
    container: CSS;
    fallback: CSS;
    image: CSS;
    text: CSS;
  };
  channelListFooterLoadingIndicator: {
    container: CSS;
  };
  channelListHeaderErrorIndicator: {
    container: CSS;
    errorText: CSS;
  };
  channelListHeaderNetworkDownIndicator: {
    container: CSS;
    errorText: CSS;
  };
  channelPreview: {
    container: CSS;
    date: CSS;
    details: CSS;
    detailsTop: CSS;
    message: CSS & {
      color: string;
      fontWeight: string;
      unreadColor: string;
      unreadFontWeight: string;
    };
    title: CSS;
  };
  closeButton: {
    container: CSS;
  };
  colors: typeof Colors;
  iconBadge: {
    container: CSS;
    icon: CSS;
    iconInner: CSS;
    unreadCount: CSS;
  };
  iconSquare: {
    container: CSS;
  };
  loadingErrorIndicator: {
    container: CSS;
    errorText: CSS;
    retryText: CSS;
  };
  loadingIndicator: {
    container: CSS;
    loadingText: CSS;
  };
  message: {
    actions: {
      button: CSS & {
        defaultBackgroundColor: string;
        defaultBorderColor: string;
        primaryBackgroundColor: string;
        primaryBorderColor: string;
      };
      buttonText: CSS & {
        defaultColor: string;
        primaryColor: string;
      };
      container: CSS;
    };
    actionSheet: {
      buttonContainer: CSS;
      buttonText: CSS;
      cancelButtonContainer: CSS;
      cancelButtonText: CSS;
      titleContainer: CSS;
      titleText: CSS;
    };
    avatarWrapper: {
      container: CSS;
      spacer: CSS;
    };
    card: {
      container: CSS;
      cover: CSS;
      footer: CSS & {
        description: CSS;
        link: CSS;
        logo: CSS;
        title: CSS;
      };
    };
    container: CSS;
    content: {
      container: CSS & {
        borderRadiusL: number;
        borderRadiusS: number;
      };
      containerInner: CSS;
      deletedContainer: CSS;
      deletedText: CSS;
      errorContainer: CSS & {
        backgroundColor: string;
      };
      // Available options for styling text: https://github.com/CharlesMangwa/react-native-simple-markdown/tree/next#styles-1
      markdown: MarkdownStyle;
      metaContainer: CSS;
      metaText: CSS;
      textContainer: CSS & {
        borderRadiusL: number;
        borderRadiusS: number;
        leftBorderColor: string;
        leftBorderWidth: number;
        rightBorderColor: string;
        rightBorderWidth: number;
      };
    };
    file: {
      container: CSS;
      details: CSS;
      icon: CSS;
      size: CSS;
      title: CSS;
    };
    gallery: {
      doubleSize: number;
      galleryContainer: CSS;
      halfSize: number;
      header: {
        button: CSS;
        container: CSS;
      };
      imageContainer: CSS;
      single: CSS;
      size: number;
      width: number;
    };
    reactionList: {
      container: CSS;
      reactionCount: CSS;
    };
    reactionPicker: {
      column: CSS;
      container: CSS;
      containerView: CSS;
      emoji: CSS;
      reactionCount: CSS;
      text: CSS;
    };
    replies: {
      container: CSS;
      image: CSS;
      messageRepliesText: CSS;
    };
    status: {
      checkMark: CSS;
      deliveredCircle: CSS;
      deliveredContainer: CSS;
      readByContainer: CSS;
      readByCount: CSS;
      sendingContainer: CSS;
      sendingImage: CSS;
      spacer: CSS;
    };
  };
  messageInput: {
    actionSheet: {
      buttonContainer: CSS;
      buttonText: CSS;
      titleContainer: CSS;
      titleText: CSS;
    };
    attachButton: CSS;
    attachButtonIcon: CSS;
    container: CSS & {
      conditionalPadding: number;
    };
    editingBoxContainer: CSS;
    editingBoxHeader: CSS;
    editingBoxHeaderTitle: CSS;
    fileUploadPreview: {
      attachmentContainerView: CSS;
      attachmentView: CSS;
      container: CSS;
      dismiss: CSS;
      dismissImage: CSS;
      filenameText: CSS;
      itemContainer: CSS;
    };
    imageUploadPreview: {
      container: CSS;
      dismiss: CSS;
      dismissImage: CSS;
      itemContainer: CSS;
      upload: CSS;
    };
    inputBox: CSS;
    inputBoxContainer: CSS;
    sendButton: CSS;
    sendButtonIcon: CSS;
    suggestions: {
      command: {
        args: CSS;
        container: CSS;
        description: CSS;
        title: CSS;
        top: CSS;
      };
      container: CSS & {
        itemHeight: number;
        maxHeight: number;
      };
      item: CSS;
      mention: {
        container: CSS;
        name: CSS;
      };
      separator: CSS;
      title: CSS;
      wrapper: CSS;
    };
    uploadProgressIndicator: {
      container: CSS;
      overlay: CSS;
    };
  };
  messageList: {
    dateSeparator: {
      container: CSS;
      date: CSS;
      dateText: CSS;
      line: CSS;
    };
    errorNotification: CSS;
    errorNotificationText: CSS;
    eventIndicator: {
      date: CSS;
      memberUpdateContainer: CSS;
      memberUpdateText: CSS;
      memberUpdateTextContainer: CSS;
    };
    listContainer: CSS;
    messageNotification: {
      container: CSS;
      text: CSS;
    };
    messageSystem: {
      container: CSS;
      dateText: CSS;
      line: CSS;
      text: CSS;
      textContainer: CSS;
    };
    typingIndicatorContainer: CSS;
  };
  spinner: CSS;
  thread: {
    newThread: CSS & {
      text: CSS;
    };
  };
  typingIndicator: {
    container: CSS;
    text: CSS & {
      color: string;
      fontSize: number;
    };
  };
};

export const defaultTheme: Theme = {
  avatar: {
    container: {},
    fallback: {},
    image: {},
    text: {},
  },
  channelListFooterLoadingIndicator: {
    container: {},
  },
  channelListHeaderErrorIndicator: {
    container: {},
    errorText: {},
  },
  channelListHeaderNetworkDownIndicator: {
    container: {},
    errorText: {},
  },
  channelPreview: {
    container: {},
    date: {},
    details: {},
    detailsTop: {},
    message: {
      color: '#767676',
      fontWeight: 'normal',
      unreadColor: '#000',
      unreadFontWeight: 'bold',
    },
    title: {},
  },
  closeButton: {
    container: {},
  },
  colors: {
    ...Colors,
  },
  iconBadge: {
    container: {},
    icon: {},
    iconInner: {},
    unreadCount: {},
  },
  iconSquare: {
    container: {},
  },
  loadingErrorIndicator: {
    container: {},
    errorText: {},
    retryText: {},
  },
  loadingIndicator: {
    container: {},
    loadingText: {},
  },
  message: {
    actions: {
      button: {
        defaultBackgroundColor: 'white',
        defaultBorderColor: 'transparent',
        primaryBackgroundColor: Colors.primary,
        primaryBorderColor: Colors.light,
      },
      buttonText: {
        defaultColor: 'black',
        primaryColor: 'white',
      },
      container: {},
    },
    actionSheet: {
      buttonContainer: {},
      buttonText: {},
      cancelButtonContainer: {},
      cancelButtonText: {},
      titleContainer: {},
      titleText: {},
    },
    avatarWrapper: {
      container: {},
      spacer: {},
    },
    card: {
      container: {},
      cover: {},
      footer: {
        description: {},
        link: {},
        logo: {},
        title: {},
      },
    },
    container: {},
    content: {
      container: {
        borderRadiusL: 16,
        borderRadiusS: 2,
      },
      containerInner: {},
      deletedContainer: {},
      deletedText: {},
      errorContainer: {
        backgroundColor: Colors.danger,
      },
      markdown: {},
      metaContainer: {},
      metaText: {},
      textContainer: {
        borderRadiusL: 16,
        borderRadiusS: 2,
        leftBorderColor: 'rgba(0, 0, 0, 0.08)',
        leftBorderWidth: 0.5,
        rightBorderColor: 'transparent',
        rightBorderWidth: 0,
      },
    },
    file: {
      container: {},
      details: {},
      icon: {},
      size: {},
      title: {},
    },
    gallery: {
      doubleSize: 240,
      galleryContainer: {},
      halfSize: 80,
      header: {
        button: {},
        container: {},
      },
      imageContainer: {},
      single: {},
      size: 120,
      width: 240,
    },
    reactionList: {
      container: {},
      reactionCount: {},
    },
    reactionPicker: {
      column: {},
      container: {},
      containerView: {},
      emoji: {},
      reactionCount: {},
      text: {},
    },
    replies: {
      container: {},
      image: {},
      messageRepliesText: {},
    },
    status: {
      checkMark: {},
      deliveredCircle: {},
      deliveredContainer: {},
      readByContainer: {},
      readByCount: {},
      sendingContainer: {},
      sendingImage: {},
      spacer: {},
    },
  },
  messageInput: {
    actionSheet: {
      buttonContainer: {},
      buttonText: {},
      titleContainer: {},
      titleText: {},
    },
    attachButton: {},
    attachButtonIcon: {},
    container: {
      conditionalPadding: 20,
    },
    editingBoxContainer: {},
    editingBoxHeader: {},
    editingBoxHeaderTitle: {},
    fileUploadPreview: {
      attachmentContainerView: {},
      attachmentView: {},
      container: {},
      dismiss: {},
      dismissImage: {},
      filenameText: {},
      itemContainer: {},
    },
    imageUploadPreview: {
      container: {},
      dismiss: {},
      dismissImage: {},
      itemContainer: {},
      upload: {},
    },
    inputBox: {},
    inputBoxContainer: {},
    sendButton: {},
    sendButtonIcon: {},
    suggestions: {
      command: {
        args: {},
        container: {},
        description: {},
        title: {},
        top: {},
      },
      container: {
        itemHeight: 50,
        maxHeight: 250,
      },
      item: {},
      mention: {
        container: {},
        name: {},
      },
      separator: {},
      title: {},
      wrapper: {},
    },
    uploadProgressIndicator: {
      container: {},
      overlay: {},
    },
  },
  messageList: {
    dateSeparator: {
      container: {},
      date: {},
      dateText: {},
      line: {},
    },
    errorNotification: {},
    errorNotificationText: {},
    eventIndicator: {
      date: {},
      memberUpdateContainer: {},
      memberUpdateText: {},
      memberUpdateTextContainer: {},
    },
    listContainer: {},
    messageNotification: {
      container: {},
      text: {},
    },
    messageSystem: {
      container: {},
      dateText: {},
      line: {},
      text: {},
      textContainer: {},
    },
    typingIndicatorContainer: {},
  },
  spinner: {},
  thread: {
    newThread: {
      text: {},
    },
  },
  typingIndicator: {
    container: {},
    text: {
      color: Colors.textGrey,
      fontSize: 14,
    },
  },
};
