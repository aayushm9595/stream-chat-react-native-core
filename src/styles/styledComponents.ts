import * as styledComponents from 'styled-components/native';

import type { Consumer, Context } from 'react';

import type { Theme } from './themeConstants';

declare module 'styled-components/native' {
  export interface DefaultTheme extends Theme {}
}

const appStyledComponents = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<
  Theme
>;

const {
  css,
  default: styled,
  ThemeProvider,
  useTheme,
  withTheme,
} = appStyledComponents;

const ThemeConsumer = appStyledComponents.ThemeConsumer as Consumer<Theme>;
const ThemeContext = appStyledComponents.ThemeContext as Context<Theme>;

export {
  css,
  styled,
  ThemeConsumer,
  ThemeContext,
  ThemeProvider as StyledComponentsThemeProvider,
  useTheme,
  withTheme,
};
