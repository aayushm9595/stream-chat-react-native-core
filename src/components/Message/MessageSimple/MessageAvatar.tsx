import React from 'react';

import { Avatar } from '../../Avatar/Avatar';

import { styled } from '../../../styles/styledComponents';

import type { ForwardedMessageProps } from './MessageContent';

import type { Alignment } from '../../../contexts/messagesContext/MessagesContext';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';

const Container = styled.View<{ alignment: Alignment }>`
  margin-left: ${({ alignment }) => (alignment === 'right' ? 8 : 0)}px;
  margin-right: ${({ alignment }) => (alignment === 'left' ? 8 : 0)}px;
  ${({ theme }) => theme.message.avatarWrapper.container.css}
`;

const Spacer = styled.View`
  height: 28px;
  width: 32px;
  ${({ theme }) => theme.message.avatarWrapper.spacer.css}
`;

export type MessageAvatarProps<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends DefaultUserType = DefaultUserType
> = ForwardedMessageProps<At, Ch, Co, Ev, Me, Re, Us> & {
  showAvatar?: boolean;
};

export const MessageAvatar = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends DefaultUserType = DefaultUserType
>(
  props: MessageAvatarProps<At, Ch, Co, Ev, Me, Re, Us>,
) => {
  const { alignment, groupStyles, message, showAvatar } = props;

  const visible =
    typeof showAvatar === 'boolean'
      ? showAvatar
      : groupStyles[0] === 'single' || groupStyles[0] === 'bottom';

  return (
    <Container alignment={alignment} testID='message-avatar'>
      {visible ? (
        <Avatar
          image={message.user?.image}
          name={message.user?.name || message.user?.id}
        />
      ) : (
        <Spacer testID='spacer' />
      )}
    </Container>
  );
};
