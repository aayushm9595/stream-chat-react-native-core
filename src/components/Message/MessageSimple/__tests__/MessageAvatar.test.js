import React from 'react';
import { cleanup, render, waitFor } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import {
  generateMessage,
  generateStaticMessage,
} from 'mock-builders/generator/message';
import { generateStaticUser } from 'mock-builders/generator/user';

import { MessageAvatar } from '../MessageAvatar';

import { defaultTheme } from '../../../../styles/themeConstants';

afterEach(cleanup);

describe('MessageAvatar', () => {
  it('should render message avatar', async () => {
    const staticUser = generateStaticUser(0);
    const message = generateMessage({
      user: { ...staticUser, image: undefined },
    });
    const {
      getByTestId,
      getByText,
      queryAllByTestId,
      queryAllByText,
      rerender,
      toJSON,
    } = render(
      <ThemeProvider theme={defaultTheme}>
        <MessageAvatar
          alignment='right'
          groupStyles={['bottom']}
          message={message}
        />
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('message-avatar')).toBeTruthy();
      expect(getByTestId('avatar-text')).toBeTruthy();
      expect(getByText(staticUser.name.charAt(0))).toBeTruthy();
    });

    rerender(
      <ThemeProvider theme={defaultTheme}>
        <MessageAvatar alignment='right' groupStyles={[]} message={message} />
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('spacer')).toBeTruthy();
      expect(queryAllByTestId('avatar-text')).toHaveLength(0);
      expect(queryAllByTestId('avatar-image')).toHaveLength(0);
    });

    const staticMessage = generateStaticMessage('hi', {
      user: staticUser,
    });

    rerender(
      <ThemeProvider theme={defaultTheme}>
        <MessageAvatar
          alignment='left'
          groupStyles={['single']}
          message={staticMessage}
        />
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('message-avatar')).toBeTruthy();
      expect(getByTestId('avatar-image')).toBeTruthy();
      expect(queryAllByText(staticUser.name.charAt(0))).toHaveLength(0);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
