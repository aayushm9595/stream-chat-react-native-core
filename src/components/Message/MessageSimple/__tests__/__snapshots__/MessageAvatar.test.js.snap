// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageAvatar should render message avatar 1`] = `
<View
  alignment="left"
  style={
    Array [
      Object {
        "marginLeft": 0,
        "marginRight": 8,
      },
    ]
  }
  testID="message-avatar"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Image
      accessibilityLabel="initials"
      onError={[Function]}
      resizeMethod="resize"
      size={32}
      source={
        Object {
          "uri": "https://images-na.ssl-images-amazon.com/images/I/51t29lLkg8L._AC_SL1000_.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 16,
            "height": 32,
            "width": 32,
          },
        ]
      }
      testID="avatar-image"
    />
  </View>
</View>
`;
